changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,203 @@
+depends on the specific implementation
+and the idioms of that implementation's language to describe APIs.
+
+## Out of Scope
+
+- Significantly changing the shape of any of our Encrypt API implementations is out of scope.
+
+## Motivation
+
+All implementations allow users to specify a plaintext length when calling Encrypt with a
+unknown-length plaintext.
+However, its behavior in the Encrypt operation is not clearly specified.
+
+Additionally, some implementations also allow a plaintext length to be specified on Encrypt with a
+known-length plaintext.
+The behavior of Encrypt in this case is not consistent between implementations.
+
+The purpose of this change is to specify exactly how this input affects the Encrypt operation.
+
+This change renames this input to `plaintext length bound` within the spec because
+the intention behind the value of this input is that
+it is a max bound of the plaintext length,
+set by the user when streaming encryption.
+If the actual plaintext length is greater than this value,
+something is wrong and the operation MUST fail.
+Otherwise, the Encrypt operation MUST pass this value to the CMM.
+
+The intention of this value is not to be a one to one passthrough value to the CMM.
+If that were the case,
+then we would be concerned with always letting users specify this value for known-length plaintexts,
+or with possibly letting users pass in a "unknown length" intent.
+We do not want to support these cases.
+Users should not be setting a `max plaintext length` value for known-length plaintexts
+with the intention of changing CMM behavior.
+Instead, users should be specifying this value in for unknown-length plaintexts in order to use CMMs
+that depend on plaintext length for their internal logic.
+
+This change additionally renames the `plaintext length` field on GetEncryptionMaterialsRequest to
+`max plaintext length` because that better explains the intent behind this field.
+This field describes the max size of the plaintext which will be encrypted using the materials
+the CMM is to generate.
+This is the current state of the behavior, so the specification should describe it as such.
+This is renamed to a different name than `plaintext length bound` in order to
+distinguish the two controls.
+
+This change does not strictly prescribe how users should express intent for the plaintext
+and a possible `plaintext length bound` through an API.
+How an API should be designed with these controls in mind depends greatly on the capabilities and
+idioms of the language in use.
+Specifically, this gets tricky for languages where the idiom is to
+accept an object represent a set of user supplied params.
+If a user supplies both a known-length plaintext and a `max plaintext length`,
+what should be done?
+It doesn't make sense to have to check for bad values in the user supplied params,
+given it's unstructured nature.
+Thus, our specification reccomends that languages SHOULD either ensure that `max plaintext length`
```

Comment:
np: `recommends`

np: `languages SHOULD either ensure ` -> `languages SHOULD ensure ` ?